library/x509_create.c

```diff
@@ -116,16 +117,21 @@ int mbedtls_x509_string_to_names( mbedtls_asn1_named_data **head, const char *na
 
         if( !in_tag && *c == '\\' && c != end )
         {
-            c++;
-
             /* Check for valid escaped characters */
-            if( c == end || *c != ',' )
+            if( ( c+1 ) == end || *( c+1 ) != ',' )
+            {
+                ret = MBEDTLS_ERR_X509_INVALID_NAME;
+                goto exit;
+            }
+            *(d++) = *c;
```

Comment:
both write and read of x509 certificate get into this function, so the parsing will be done correct